e2e/test/scenarios/table-editing/table-editing.cy.spec.ts

```diff
@@ -499,6 +499,70 @@ describe("scenarios > table-editing", () => {
       cy.findAllByTestId("row-edit-icon").should("exist");
     });
   });
+
+  describe("table editing bugs", () => {
+    it("WRK-907: should not allow to create new values for FK fields", () => {
+      cy.intercept("GET", `/api/table/${ORDERS_ID}/query_metadata`).as(
+        "getOrdersTable",
+      );
+      const NON_EXISTING_ID = "999999";
+      cy.intercept(
+        "GET",
+        `/api/field/${ORDERS.USER_ID}/search/${ORDERS.USER_ID}?value=${NON_EXISTING_ID}&limit=20`,
+      ).as("getFieldValues");
+      cy.intercept(
+        "GET",
+        `/api/field/${PRODUCTS.CATEGORY}/search/${PRODUCTS.CATEGORY}?value=${NON_EXISTING_ID}&limit=20`,
+      ).as("getCategoryValues");
+
+      cy.visit(`/browse/databases/${SAMPLE_DB_ID}/tables/${ORDERS_ID}/edit`);
+
+      cy.wait("@getOrdersTable");
+
+      cy.findByTestId("new-record-button").click();
+
+      H.modal().within(() => {
+        cy.findByTestId("User ID-field-input").click();
+        cy.realType(NON_EXISTING_ID);
+      });
+
+      cy.wait("@getFieldValues");
+
+      H.popover().within(() => {
+        cy.findByText("Nothing found").should("be.visible");
+        cy.findByText(`Add option: ${NON_EXISTING_ID}`).should("not.exist");
+      });
+
+      H.modal().within(() => {
+        cy.findByText("Cancel").click();
+      });
```

Comment:
nit: if within callback contains only 1 LOC

```suggestion
      H.modal().findByText("Cancel").click();
```